common/lib/xmodule/xmodule/js/src/video/08_video_auto_advance_control.js

```diff
@@ -0,0 +1,129 @@
+                this.state = state;
+                this.state.videoAutoAdvanceControl = this;
+                this.initialize();
+
+                return $.Deferred().resolve().promise();
+            };
+
+            AutoAdvanceControl.prototype = {
+                template: HtmlUtils.interpolateHtml(
+                    HtmlUtils.HTML([
+                        '<button class="control auto-advance" aria-disabled="false" title="',
+                        '{autoAdvanceText}',
+                        '">',
+                        '<span class="label" aria-hidden="true">',
+                        '{autoAdvanceText}',
+                        '</span>',
+                        '</button>'].join('')),
+                    {
+                        autoAdvanceText: gettext('Auto-advance')
+                    }
+                ).toString(),
+
+                destroy: function() {
+                    this.el.off({
+                        click: this.onClick
+                    });
+                    this.el.remove();
+                    this.state.el.off({
+                        ready: this.autoPlay,
+                        ended: this.autoAdvance,
+                        destroy: this.destroy
+                    });
+                    delete this.state.videoAutoAdvanceControl;
+                },
+
+                /** Initializes the module. */
+                initialize: function() {
+                    var state = this.state;
+
+                    this.el = $(this.template);
+                    this.render();
+                    this.setAutoAdvance(state.auto_advance);
+                    this.bindHandlers();
+
+                    return true;
+                },
+
+                /**
+                 * Creates any necessary DOM elements, attach them, and set their,
+                 * initial configuration.
+                 * @param {boolean} enabled Whether auto advance is enabled
+                 */
+                render: function() {
+                    this.state.el.find('.secondary-controls').prepend(this.el);
+                },
+
+                /**
+                 * Bind any necessary function callbacks to DOM events (click,
+                 * mousemove, etc.).
+                 */
+                bindHandlers: function() {
+                    this.el.on({
+                        click: this.onClick
+                    });
+                    this.state.el.on({
+                        ready: this.autoPlay,
+                        ended: this.autoAdvance,
+                        destroy: this.destroy
+                    });
+                },
+
+                onClick: function(event) {
+                    var enabled = !this.state.auto_advance;
+                    event.preventDefault();
+                    this.setAutoAdvance(enabled);
+                    this.el.trigger('autoadvancechange', [enabled]);
```

Comment:
Why do need to pass `enabled` this in array ?

Reply:
It seems to be jquery's syntax:
http://api.jquery.com/trigger/
> An array of arguments can also be passed to the .trigger() call, and these parameters will be passed along to the handler as well following the event object. As of jQuery 1.6.2, single string or numeric argument can be passed without being wrapped in an array. 

Most of the code is using the array syntax for single arguments, so I used it too.